### DISCUSSION TITLE
Sonoma Photoanalysis scores: no meaningful changes

### DISCUSSION THREAD

**Discussion by cfc62** [RhetTbull/osxphotos#1309] [General]
While scores may have changed ever so slightly, for example from 0.493204 to 0.493508, all of the findings from my previous analysis of photoanalysis scores remain completely intact. 

Not saying there weren't some (extremely) minor tweaks but clearly nothing of any note has changed in Sonoma as far as scoring goes. 

I do see some scores for mediaanalysis in the schema, but nothing that looks horribly interesting at first glance. 

I would like to start creating albums of images with high and low scores to visually understand what the scores represent, the last time I tried this I had issues and I'm not sure I see updates to osxphotos that would improve this, is that correct?

Thanks again! Wish something more interesting had come out of this update as clearly there's a lot more going on in Sonoma and the scoring that exists today seems less than stellar.

**Comment by RhetTbull** [comment]
> I would like to start creating albums of images with high and low scores to visually understand what the scores represent, the last time I tried this I had issues and I'm not sure I see updates to osxphotos that would improve this, is that correct?

I don't recall which issues you had but the following should work:

`osxphotos query --quiet --not-shared --query-eval "photo.score.overall > 0.8" --add-to-album "High Scores"`

And for low scores (obviously adjust the scores to your liking:

`osxphotos query --quiet --not-shared --query-eval "photo.score.overall < 0.3" --add-to-album "Low Scores"`

**Reply by cfc62** [reply]
IIRC (and I may not) I was using a command you mentioned was leveraging AppleScript and it was taking hours as a result; when I bailed out I had a corrupted library, which I could fix without problem.

**Reply by RhetTbull** [reply]
Ah, ok. One option may be to limit by date:

`osxphotos query --quiet --not-shared --query-eval "photo.score.overall > 0.8" --add-to-album "High Scores" --from-date "2021-01-01" --to-date "2022-01-01`

for example will only process photos from 2021. You could then adjust the dates to grab a year's worth at a time.

I'm working on some new code that allows me to use the native API without using AppleScript and that should be much more robust -- probably still a few months out from release though.

**Reply by cfc62** [reply]
Thanks - I obviously don't need hundreds of examples, a couple dozen or so should be enough to... _illustrate_ (I am so sorry) what the scores represent...

**Reply by cfc62** [reply]
It's working well enough, I'm able to use the distributions from the analysis to only grab a few dozen for each score, both the high and low outliers.

Thanks!

**Reply by cfc62** [reply]
Unfortunately, it was working and now it *seems* to work but the created albums are empty. I'll poke around a bit, reboot, all that jazz...

**Reply by cfc62** [reply]
Had to rebuild/restore the library, which apparently clears all the scores as well. So I can't continue until the scores are re-generated. 

Something of a Hobson's choice here - I can press forward with running these queries, but possibly get into a state where the library is unrecoverable, or just not bother with it. I'm torn...

From what I have been able to see so far:

**Low curation scores** (below 0.3) seem to represent:
- Pictures of screens (such as airport departure boards), screenshots
- Pictures of documents, papers
- Pictures of signs
- Pictures of boxes, labels, tickets
- A fairly high proportion of self-made time lapse videos (not generated on iPhone)
- Some iPhone time lapse videos
- Pictures of logos, emblems, badges
- The remainder (perhaps 35%) are perfectly fine photos

**High curation scores** (above 0.7) seem to include:
- Higher-quality landscape photos, especially with blue skies and/or sunset/sunrise colors
- Blue water, beaches
- People, especially with bokeh
- Flowers, mainly with bokeh
- Uncomplicated shots of architecture (e.g., shots that have a clear emphasis to them), especially with blue sky backdrop
- There are definitely a few stinkers in this batch, but a smaller percentage as compared to the misses in the low curation scores

Oh, there are VERY similar photos with one being curated low, and another being curated very highly. 

Do you want to blindly remove all low curation score photos, and keep all of the high ones? **_Bad idea._**  Some of the lower curation score photos are going to be important to you - maybe a ticket stub from a memorable event, a picture of a famous person's gravestone, could be anything. Some of the higher curation scores are also puzzling. Out of focus, poor lighting... 

I want to collect more high and low overall scoring photos before coming to any sort of conclusion. Only a small number of photos get curated, so it's easier to see what the ends of the spectrum represent. Conversely, every asset gets an overall score so that's more involved. 

I do have the very lowest behavioral score photos, and I have no earthly idea what this represents as of yet. I'll have to compare it against the highest behavior score assets, which is what I was trying to do until I realized there were no results coming out of the query after the library was repaired/restored.

**Comment by cfc62** [comment]
Found it:

<img width="932" alt="image" src="https://github.com/RhetTbull/osxphotos/assets/70529418/ac6fa337-b253-4a07-8635-4346cef2205c">
